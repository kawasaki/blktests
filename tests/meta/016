#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2024 Western Digital Corporation or its affiliates.
#
# Test repeated test() run with set_conditions()

. tests/meta/rc

DESCRIPTION="repeat test()"

declare cond_set_index

set_conditions() {
	local index=$1

	if [[ -z $index ]]; then
		echo 2
		return
	fi

	cond_set_index=$index
	COND_DESC="condition set $index"
}

test() {
	echo "Running ${TEST_NAME}"
	echo "condition set $cond_set_index" >> "$FULL"
	echo "Test complete"
}
