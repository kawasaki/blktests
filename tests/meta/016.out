Running meta/016
Test complete
